pyccel/parser/syntax/basic.py

```diff
@@ -1,5 +1,12 @@
 # coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
 
+"""
+This module contains the Base class for all objects in Pyccel.
```

Comment:
This is not true. There are lots of classes in pyccel which do not inherit from this class.

I assume you got the message from the docstring for this class? Unfortunately this class doesn't seem to have been touched since 2017.

@saidctb do you know what it does? Do all classes needed for parsing the headers inherit from it?

Reply:
it's actually useless now, its properties are not used, but all the classes that  define the grammar nodes inherit from it

Reply:
So then no docstring and it needs removing in the future?

Reply:
@mostafaMamoni This docstring can be removed as it is incorrect